src/states/node.rs

```diff
@@ -1516,98 +1417,19 @@ impl Node {
         {
-            info!("{}: Init new Parsec, genesis = {:?}", self, genesis_info);
-
-            let full_id = self.full_id.clone();
-            let genesis_ver = *genesis_info.first_info.version();
-            let consensus_mode = parsec::ConsensusMode::Single;
-
-            #[cfg(not(feature = "mock_parsec"))]
-            let parsec = if genesis_info.first_info.members().contains(self.id()) {
-                Parsec::from_genesis(full_id, &genesis_info.first_info.members(), consensus_mode)
-            } else {
-                Parsec::from_existing(
-                    full_id,
-                    &genesis_info.first_info.members(),
-                    &genesis_info.latest_info.members(),
-                    consensus_mode,
-                )
-            };
-
-            #[cfg(feature = "mock_parsec")]
-            let parsec = {
-                let section_hash = *genesis_info.first_info.hash();
-
-                if genesis_info.first_info.members().contains(self.id()) {
-                    Parsec::from_genesis(
-                        section_hash,
-                        full_id,
-                        &genesis_info.first_info.members(),
-                        consensus_mode,
-                    )
-                } else {
-                    Parsec::from_existing(
-                        section_hash,
-                        full_id,
-                        &genesis_info.first_info.members(),
-                        &genesis_info.latest_info.members(),
-                        consensus_mode,
-                    )
-                }
-            };
-
-            let _ = self.parsec_map.insert(genesis_ver, parsec);
-
-            return Ok(true);
-        }
-        Ok(false)
-    }
-
-    fn init_chain(&mut self) {
-        if let Some(ref genesis_info) = self.gen_pfx_info {
-            self.chain = Chain::with_gen_info(
-                self.chain.min_sec_size(),
-                *self.full_id.public_id(),
-                genesis_info.clone(),
+            let _ = entry.insert(create_parsec(self.full_id.clone(), &self.gen_pfx_info));
+            info!(
+                "{}: Init new Parsec, genesis = {:?}",
+                self, self.gen_pfx_info
             );
         }
     }
 
-    fn handle_node_approval(&mut self, gen_info: GenesisPfxInfo) -> Result<(), RoutingError> {
-        if self.gen_pfx_info.is_some() {
-            log_or_panic!(LogLevel::Warn, "{} Received duplicate NodeApproval.", self);
-            return Ok(());
-        }
-
-        self.resource_prover.handle_approval();
-        info!(
-            "{} Resource proof challenges completed. This node has been approved to join the \
-             network!",
-            self
-        );
-
-        self.gen_pfx_info = Some(gen_info.clone());
-        if self.init_parsec()? {
-            self.init_chain();
-        }
-
-        self.poke_timer_token = Some(self.timer.schedule(Duration::from_secs(POKE_TIMEOUT_SECS)));
```

Comment:
For my info, what does this poke_timer_token represent? (I see you kept it in `Node`, but not in `ResourceProving` state)

Reply:
"Poke" is a message send from a newly joined node to a random member of the section to nudge them to send the first parsec gossip. A node upon receiving the poke will immediately respond by sending parsec request back. It isn't in `ProvingNode` because there is no `Parsec` yet.